### PR TITLE
julia 1.7.2

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by carlocab** [comment]
Patches need updating.

**Comment by cho-m** [comment]
Blocked until #96067 is merged.

Also will need to rebase either this or #95053, since in `libgit2` PR `julia` needs an extra patch to fix `libgit2 >= 1.4`.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**
